jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppClassLoader.java

```diff
@@ -290,14 +293,20 @@ public void addJars(Resource libs)
 
             try (Stream<Path> streamEntries = Files.list(dir))
             {
-                streamEntries
+                List<URI> entries = streamEntries
                     .filter(Files::isRegularFile)
                     .filter(this::isFileSupported)
                     .sorted(Comparator.naturalOrder())
                     .map(Path::toUri)
+                    .filter(new UniqueURIPredicate(getURLs())) // ignore URLs that are already present in classloader
                     .map(URIUtil::toJarFileUri)
-                    .map(_resourceFactory::newResource)
-                    .forEach(this::addClassPath);
+                    .toList();
+
+                for (URI uri: entries)
```

Comment:
Why building a list of URIs then iterating it instead of the previous `map()` and `forEach()`?